### ISSUE TITLE
Kuma not able to get transformed request from Azure API Management

### ISSUE THREAD

**Issue by R8iKKo** [louislam/uptime-kuma#4897]
### ⚠️ Please verify that this question has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

Hi, I am trying to develop custom API for Ansible REST API and Azure API Management. I kinda figured everything out, but when it came to the final step - which was the whole point of doing this. 

But after I implemented some outbound transformations in Azure APIM, Kuma started to throw `500` codes for the API. When I remove the set-variable, I am able to get `200` from the API, but without the important json query that tells the policy whether to set `200` or `400` response. 

But my main point is, it's only Kuma that throws the `500` error. Postman is pretty OK with the output and works fine, Azure test module in API also prints everything fine, including trace. 

<details><summary>Here is my policy code for the API:</summary>
<p>

```
<policies>
    <!-- Throttle, authorize, validate, cache, or transform the requests -->
    <inbound>
        <base />
    </inbound>
    <backend>
        <base />
    </backend>
    <!-- Customize the responses -->
    <outbound>
        <base />
        <set-header name="Content-Type" exists-action="override">
            <value>application/json</value>
        </set-header>
        <set-variable name="statusValue" value="@(context.Response.Body.As<JObject>()["summary_fields"]["last_job"]["status"].ToString().Trim().ToLower())" />
        <choose>
            <when condition="@(context.Variables.GetValueOrDefault<string>("statusValue") == "successful")">
                <return-response>
                    <set-status code="200" reason="OK" />
                    <set-body template="liquid">
{ 
"status": "Workflow run went OK"
}
                    </set-body>
                </return-response>
            </when>
            <when condition="@(context.Variables.GetValueOrDefault<string>("statusValue") == "failed")">
                <return-response>
                    <set-status code="400" reason="Workflow Failed" />
                    <set-body template="liquid">
{ 
"status": "Workflow run failed"
}
                    </set-body>
                </return-response>
            </when>
            <!--<otherwise>
                <return-response>
                    <set-status code="500" reason="Request Failed - Bad API" />
                </return-response>
            </otherwise>-->
        </choose>
    </outbound>
    <!-- Handle exceptions and customize error responses  -->
    <on-error>
        <base />
    </on-error>
</policies>
```

</p>
</details> 

And basically when I remove these policies, Kuma is able to retrieve 200, but its full json with a lot a data, which is useless.
I also tried JSON Query with Kuma, but that doesn't work either. 

My point is, that when I send request with Kuma I get `500` and error message from Azure AppInsights below.

When I sent the same request from Azure or Postman, I get `200` and desired response.
So is it possible, that Kuma somehow cannot get transformed json? Or where can be the problem, obviously the policy code is pretty basic hence I totally don't know this XML language, but if it was fundamentally wrong, wouldn't also podman throw `500`?

Thank you for every advice or notice.

### 📝 Error Message(s) or Log

```
Expression evaluation failed. The message body is not a valid JSON. Unexpected character encountered while parsing value: <. Path '', line 0, position 0.
   at Newtonsoft.Json.JsonTextReader.ParseValue()
   at Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.Models.SyncOnlyDepthTrackingJsonTextReader.Read() in D:\a\1\s\Proxy\Gateway.Pipeline.IO\SyncOnlyDepthTrackingJsonTextReader.cs:line 62
   at Newtonsoft.Json.Linq.JObject.Load(JsonReader reader, JsonLoadSettings settings)
   at Gateway.Pipeline.IO.JsonConverter.Convert[T](Stream stream, Encoding encoding, ILog log, Object conversionSettings) in D:\a\1\s\Proxy\Gateway.Pipeline.IO\MessageConverters\JsonConverter.cs:line 57
   at Microsoft.WindowsAzure.ApiManagement.Proxy.Gateway.MessageBody.As[T](Boolean preserveContent) in D:\a\1\s\Proxy\Gateway.Pipeline.IO\MessageBody.cs:line 83 
```

### 🐻 Uptime-Kuma Version

1.23.13

### 💻 Operating System and Arch

Azure B1 ASP

### 🌐 Browser

Safari 17.5 (19618.2.12.11.6)

### 🖥️ Deployment Environment

- Runtime:  Docker
- Database: 
- Filesystem used to store the database on: Linux ASP
- number of monitors: 3

**Comment by CommanderStorm** [comment]
I have never heard of that, nor worked with it.

The error message states that the first char is `<`...
=> have you checked that comments like `<!--<otherwise>` are supported?

**Comment by R8iKKo** [comment]
Hi, not sure if I understand correctly, but yes, <!--<otherwise> comment is native to this horrible Azure XML language.. I can remove those, but wouldn't this also be shown with Postman for example? I still quite don't understand where is the difference between Postman and Kuma processing response body.

**Comment by CommanderStorm** [comment]
True, those would be the same, but maybe that case is triggered due to different request headers like `Accepts` (See #4491)
Have you specified all relevant headers?
Maybe azure does some magic to requests? (You would need to ask in their forum or support. We spcecify [this](https://github.com/louislam/uptime-kuma/blob/cf2d603e277d18e96906ac36743ec4e7d58b8aad/server/model/monitor.js#L480))

I am not quite sure if the outgoing body from azure or the incoming body to azure is causing the error you are seeing. Please recheck if the `Content-Type` and body content is set in postman and kuma identically.

**Comment by CommanderStorm** [comment]
You can likely also debug the policy via https://learn.microsoft.com/de-de/azure/api-management/trace-policy

**Comment by R8iKKo** [comment]
Ok thank you for your advices. 

A revisioned the code and made some changes and came to even more weird outcome. The body was probably indeed in invalid JSON, because now I changed the policy to take whole body in json and extract the object value I needed. In trace I see that all these operations went well, taking full body, then transforming to JSON and then extracting the particular value and successfully did the condition. 

BUT

In podman I got 200 again, no problem, I also added Content-type: application/json to response header. 
Kuma again throws 500, but now I don't see any error regarding the JSON body or anything else in the Azure log. Now it just shows that backend answers 200 and back from GW to Kuma it throws 500. Without any obvious reason as before. 

So now I am kinda blind with what is the reason of 500 response from Kuma. Is there any place I can see it directly in Kuma? In log I only see that it returned 500 but no error message. 

Thank you.

**Comment by CommanderStorm** [comment]
And the `Accepts` header?

**Comment by R8iKKo** [comment]
Ok I got you now, I have set same Accept header as Kuma for Podman and it is indeed the problem. I now get the same response, so I changed the policy to send response in text/html and I get 200 on Kuma too. The problem now is, that with text I am not able to perform the condition that takes the value from the json response and decide whether to send 200 or 400.  Even when the status is failed now, I get 200 with full html page response :/

**Comment by R8iKKo** [comment]
Ok so I got it fixed! It was actually stupid easy... :D 

All I had to add was Content-Type: application/json to Kuma headers. That's it.

Now I am able to get transformed json responses.

Somehow it doesn't work when you specify Accept headers in request like: */* or just text/html or even combined application/json,text/html. That didn't work even with Postman.

But I sorted it out and now it works just as it should with Kuma.

Still thank you very much with every advice and super fast responses. I really appreciate it.
